### ISSUE TITLE
Lights: Removing BaseObject3D.addLight() and adding RajawaliScene.addLight()

### ISSUE THREAD

**Issue by MasDennis** [Rajawali/Rajawali#816]
I will be changing this because it doesn't make sense and it is hard to manage.
Currently lights are added to `BaseObject3D` which makes it hard to manage with shared materials. I've checked other 3D engines like Irrlicht, Ogre3D, Away3d and Three.js and they all add lights to the scene object which makes much more sense.

This will affect your scene graph work @jwoolston so you input would be much appreciated.

We need to think about how we deal with materials and lights. We already discussed that we should have a minimum number of lights per material. 
When there are more lights used than this minimum it will cause the shaders to recompile (if deferred rendering isn't used).

To summarise:
- [ ] remove `setLights()` from `BaseObject3D`
- [ ] remove `getLights()` from `BaseObject3D`
- [ ] remove `getLight()` from `BaseObject3D`
- [ ] add `addLight()` to `RajawaliScene`
- [ ] add `removeLight()` to `RajawaliScene`
- [ ] add `getLights()` to `RajawaliScene`
- [ ] add the minimum number of lights to `AMaterial`
- [ ] add logic to `AMaterial` that triggers a program recompile when lights are added > minimum
- [ ] set the lights on each material in the render loop.
